pulpcore/app/viewsets/base.py

```diff
@@ -349,9 +348,10 @@ def get_queryset(self):
                 filters[lookup] = self.kwargs[key]
             qs = qs.filter(**filters)
 
-        permission_name = getattr(self, "queryset_filtering_required_permission", None)
-        if permission_name:
-            qs = get_objects_for_user(self.request.user, permission_name, qs)
+        if getattr(self, "request", None):
+            for permission_class in self.get_permissions():
+                if hasattr(permission_class, "scope_queryset"):
+                    qs = permission_class.scope_queryset(self, qs)
```

Comment:
This results in the intersection of all PermissionClass scopes, right? Should we do:
```suggestion
        if getattr(self, "request", None):
            new_qs = qs.none()
            replace = False
            for permission_class in self.get_permissions():
                if hasattr(permission_class, "scope_queryset"):
                    new_qs |= permission_class.scope_queryset(self, qs)
                    replace=True
            if replace:
                qs = new_qs
```

So when there is at least one PermissionClass with scoping we combine all of them?